### PR TITLE
1187 fix: artik05x `make download all` fails

### PR DESCRIPTION
**Pull request by pholat**
* There is need to check whether it's mac or linux in artik05_download.sh - as binaries differ.
* if there is no `sum` variable then partition_gen.sh will faill in first eval, without `set -e` it will continue to assign just base address without sum at all which will end in useless download onto artik. Everything will be loaded to 40000000 address.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by pholat** [comment]
Sure, changed. If commit rules could be added it would be apreciated. 
I've added another fix, so I'll redo the PR.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@pholat Thank you for contribution.
Unfortunately we don't provide commit rules, internally we shared.
```
Don't use past tense.
Please let us know what and where you modify on commit title.
Please fill commit description to know what you do in detail.
Please keep short commit title to beautiful on git commit --oneline.
```

Usaually where do you find those kind rules? on wiki? or on home (first page of web)?
We will add them.

Thank you.

**Comment by sunghan-chang** [comment]
I added [[commit rules]](https://github.com/Samsung/TizenRT/wiki/Documentations#coding-rules) on our wiki.

File: external/iotjs/deps/jerry/CMakeLists.txt [hunk 1]
```diff
@@ -139,7 +139,7 @@ macro(jerry_add_compile_warnings)
   foreach(_warning ${ARGV})
     jerry_add_compile_flags(-W${_warning})
     if(USING_GCC)
-      jerry_add_compile_flags(-Werror=${_warning})
+        jerry_add_compile_flags(-Werror=${_warning} -Wno-implicit-fallthrough)
```
Comment (sunghan-chang) [inline external/iotjs/deps/jerry/CMakeLists.txt]: @haesik @hs0225 Can we merge this and above change?

**Pull request merged**